### PR TITLE
Donor form

### PR DESCRIPTION
**Pull request by geralddevelop**
Data schema is as follow

```
{
    "dog-name": "sdadaasd",
    "dog-breed": "aidi",
    "dog-birthday": "2024-03-19T16:00:00.000Z",
    "dog-sex": "yes",
    "dog-weight": "23",
    "dog-blood-type": "dea1.1-positive",
    "dog-blood-transfusion-status": "yes",
    "dog-pregnant-status": "no",
    "dog-last-heartworm-vaccination": "2024-03-08T16:00:00.000Z",
    "dog-last-donation": "2024-03-12T16:00:00.000Z",
    "dog-preferred-vets": [
        "vet-b",
        "vet-a"
    ],
    "name": "asd assda",
    "mobile": "9238433432",
    "email": "geralddevelop@gmail.com",
    "country-based": "yes"
}
```
Screenshot 2024-03-10 at 8 51 34 PM
Screenshot 2024-03-10 at 8 51 57 PM

- added stepper tho it will need work :(
- split forms into parts
- mock otp flow with verify code

### CODE REVIEW

File: src/app/example/petForm.tsx [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+            {
+              label: "No",
+              value: "no",
+            },
+          ]}
+        />
+
+        <BarkFormDatetimeInput
+          form={form}
+          label="When was it’s last blood donation? (DD/MM/YYYY)"
```
Comment (danui) [inline src/app/example/petForm.tsx:159]: The label is incorrect

File: src/app/example/petForm.tsx [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+          name="dog-last-donation"
+          description="If applicable"
+        />
+
+        <BarkFormCheckboxes
```
Comment (danui) [inline src/app/example/petForm.tsx:171]: The choice of preferred vet should be singular.

File: src/app/example/donorForm.tsx [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+      {complete ? (
+        <div className="text-center">
+          <BarkH4>
+            Thank you for your information, your account has been created.
+            You&apos;ll be directed to your account shortly. If not, please log
+            in at{" "}
+            <span className="hover:underline">
+              <a target="_blank" href="https://www.pawtal.barkbank.co">
+                www.pawtal.barkbank.co
+              </a>
```
Comment (danui) [inline src/app/example/donorForm.tsx:69]: How about providing a link to the User Login page at `/user/login`?

File: public/checkMail.svg
Comment (danui) [inline public/checkMail.svg:1]: nit: I'm would like to standardise the file and directory names to use kabab case. In this case it'll be `check-mail.svg`. Ditto for the others donor-form.tsx, owner-form.tsx, pet-form.tsx

File: src/app/example/petForm.tsx (hunk 2 above)
Comment (geralddevelop) [reply src/app/example/petForm.tsx:171]: should I change this to radio button then? Cause the figma design is checkboxes

File: src/app/example/petForm.tsx (hunk 1 above)
Comment (geralddevelop) [reply src/app/example/petForm.tsx]: ok but input for heartworm vaccination and last donation seem to be removed from figma so I will remove these fields as well.

**Comment by geralddevelop** [comment]
@danui all should be done please review again thanks!.

**Review by danui — APPROVED** [review]

**Pull request merged**